llvm/lib/Support/RISCVISAInfo.cpp

```diff
@@ -854,6 +854,81 @@ RISCVISAInfo::parseArchString(StringRef Arch, bool EnableExperimentalExtension,
                              "string must be lowercase");
   }
 
+  bool IsProfile = Arch.starts_with("rvi") || Arch.starts_with("rva") ||
+                   Arch.starts_with("rvb") || Arch.starts_with("rvm");
+  std::string NewArch;
```

Comment:
Please rename NewArch to ArchStorage.  Specifically, we end up with a StringRef bound to this storage, and having it go out of scope before that StringRef would be a use-after-free.